src/Cake.Core/ICakeEngine.cs

```diff
@@ -55,9 +55,9 @@ public interface ICakeEngine
         /// </summary>
         /// <param name="context">The context.</param>
         /// <param name="strategy">The execution strategy.</param>
-        /// <param name="target">The target to run.</param>
+        /// <param name="settings">The execution settings.</param>
         /// <returns>The resulting report.</returns>
-        Task<CakeReport> RunTargetAsync(ICakeContext context, IExecutionStrategy strategy, string target);
+        Task<CakeReport> RunTargetAsync(ICakeContext context, IExecutionStrategy strategy, ExecutionSettings settings);
```

Comment:
Could we keep the old method and obsolete that? Don’t think much will break due to this, but still.

Reply:
No, sadly not possible since it then would rely on side-effects.